pkg/capi/gcp.go

```diff
@@ -0,0 +1,321 @@
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+	yaml "sigs.k8s.io/yaml"
+)
+
+const (
+	gcpMachineTemplateNamePrefix = "gcp-machine-template-"
+	OnHostMaintenanceTerminate   = "Terminate"
+	OnHostMaintenanceMigrate     = "Migrate"
+)
+
+var (
+	clusterName            string
+	cl                     client.Client
+	gcpMachineTemplateName string
+)
+
+var _ = Describe("Cluster API GCP MachineSet", framework.LabelCAPI, framework.LabelDisruptive, Ordered, func() {
+	var gcpMachineTemplate *gcpv1.GCPMachineTemplate
+	var machineSet *clusterv1.MachineSet
+	var mapiMachineSpec *mapiv1.GCPMachineProviderSpec
+	var ctx context.Context
+	var platform configv1.PlatformType
+	var clusterName string
+	var err error
+
+	BeforeAll(func() {
+		cl, err = framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred(), "Failed to create Kubernetes client for test")
+		komega.SetClient(cl)
+		ctx = framework.GetContext()
+		platform, err = framework.GetPlatform(ctx, cl)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get platform")
+		if platform != configv1.GCPPlatformType {
+			Skip("Skipping GCP E2E tests")
+		}
+		oc, _ := framework.NewCLI()
+		framework.SkipIfNotTechPreviewNoUpgrade(oc, cl)
+
+		infra, err := framework.GetInfrastructure(ctx, cl)
+		Expect(err).NotTo(HaveOccurred(), "Failed to get cluster infrastructure object")
+		Expect(infra.Status.InfrastructureName).ShouldNot(BeEmpty(), "infrastructure name was empty on Infrastructure.Status.")
+		clusterName = infra.Status.InfrastructureName
+
+		framework.CreateCoreCluster(ctx, cl, clusterName, "GCPCluster")
+		mapiMachineSpec = getGCPMAPIProviderSpec(cl)
+	})
+
+	BeforeEach(func() {
+		gcpMachineTemplateName = fmt.Sprintf("%s%s", gcpMachineTemplateNamePrefix, framework.GenerateRandomString(6))
+	})
+
+	AfterEach(func() {
+		if platform != configv1.GCPPlatformType {
+			// Because AfterEach always runs, even when tests are skipped, we have to
+			// explicitly skip it here for other platforms.
+			Skip("Skipping GCP E2E tests")
+		}
+		framework.DeleteCAPIMachineSets(ctx, cl, machineSet)
+		framework.WaitForCAPIMachineSetsDeleted(ctx, cl, machineSet)
+		framework.DeleteObjects(ctx, cl, gcpMachineTemplate)
+	})
+	DescribeTable("should be able to run a machine with disk types", framework.LabelCAPI, framework.LabelDisruptive,
+		func(expectedDiskType gcpv1.DiskType) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			gcpMachineTemplate = createGCPMachineTemplate(mapiProviderSpec)
+			gcpMachineTemplate.Spec.Template.Spec.RootDeviceType = &expectedDiskType
+			if err := cl.Create(ctx, gcpMachineTemplate); err != nil && !apierrors.IsAlreadyExists(err) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			machineSet, _ = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset-77825",
```

Comment:
Should we use the generated name also here?

Reply:
We usually use the testcase ID to get to debug quickly to make sure functionality is working for CI failures hence using the number , which will be unique as well .

Reply:
Got it, thanks